e2e/specs/patientSearch.spec.ts

```diff
@@ -9,45 +9,71 @@ test.beforeEach(async ({ api }) => {
   patient = await generateRandomPatient(api);
 });
 
-test('should be able to search patients by identifier', async ({ page }) => {
+test('Search patient by patient identifier', async ({ page, api }) => {
   // extract details from the created patient
   const openmrsIdentifier = patient.identifiers[0].display.split('=')[1].trim();
   const firstName = patient.person.display.split(' ')[0];
   const lastName = patient.person.display.split(' ')[1];
-
   const homePage = new HomePage(page);
-  await homePage.goto();
 
-  await homePage.searchPatient(openmrsIdentifier);
+  await test.step('When I visit the home page', async () => {
+    await homePage.goto();
+  });
+
+  await test.step('And I enter a valid patient identifier into the search field', async () => {
+    await homePage.searchPatient(openmrsIdentifier);
+  });
+
+  await test.step('Then I should see only the patient with the entered identifier', async () => {
+    await expect(homePage.floatingSearchResultsContainer()).toHaveText(/1 search result/);
+    await expect(homePage.floatingSearchResultsContainer()).toHaveText(new RegExp(firstName));
+    await expect(homePage.floatingSearchResultsContainer()).toHaveText(new RegExp(lastName));
+    await expect(homePage.floatingSearchResultsContainer()).toHaveText(new RegExp(openmrsIdentifier));
+  });
 
-  await expect(homePage.floatingSearchResultsContainer()).toHaveText(/1 search result/);
-  await expect(homePage.floatingSearchResultsContainer()).toHaveText(new RegExp(firstName));
-  await expect(homePage.floatingSearchResultsContainer()).toHaveText(new RegExp(lastName));
+  await test.step('When I click on the patient', async () => {
+    await homePage.clickOnPatientResult(firstName);
+  });
 
-  await homePage.clickOnPatientResult(firstName);
-  await expect(homePage.page).toHaveURL(
-    `${process.env.E2E_BASE_URL}/spa/patient/${patient.uuid}/chart/Patient Summary`,
-  );
+  await test.step("Then I should be in the patient's chart page", async () => {
+    await expect(homePage.page).toHaveURL(
+      `${process.env.E2E_UI_BASE_URL}patient/${patient.uuid}/chart/Patient Summary`,
```

Comment:
In my view, it's preferable to use the `E2E_UI_BASE_URL` variable. We don't require that the frontend be served from `/spa` (just like the app isn't required to be installed at `/openmrs`) and it's probably more broadly useful if we can override this so that implementations that opt to use a different web address can still run the e2e tests against their configuration.

Reply:
https://issues.openmrs.org/browse/O3-2125